aten/src/ATen/native/mkldnn/Conv.cpp

```diff
@@ -167,15 +175,18 @@ Tensor mkldnn_convolution(
   c10::MaybeOwned<Tensor> bias_maybe_owned = at::borrow_from_optional_tensor(bias_opt);
   const Tensor& bias = *bias_maybe_owned;
 
-  if (input.scalar_type() == ScalarType::BFloat16) {
+  if (input_t.scalar_type() == ScalarType::BFloat16) {
     TORCH_CHECK(mkldnn_bf16_device_check(),
         "mkldnn_convolution: bf16 path needs the cpu support avx512bw, avx512vl and avx512dq");
   }
 
-  check_shape_forward(input, weight, bias, padding, stride, dilation, groups);
+  check_shape_forward(input_t, weight_t, bias, padding, stride, dilation, groups);
 
-  bool is_channels_last = input.suggest_memory_format() == at::MemoryFormat::ChannelsLast;
+  bool is_channels_last = mkldnn_conv_use_channels_last(input_t, weight_t);
+  auto memory_format = mkldnn_convolution_memory_format(input_t.ndimension(), is_channels_last);
 
+  auto input = input_t.is_mkldnn() ? input_t : input_t.contiguous(memory_format);
```

Comment:
`.contiguous()` here is a duplicate, both `input` and `weight` has already been converted to contiguous in `Convolution.cpp` as
```
check_input_same_type_as_parameters(input, weight, bias, backend);
      if (!input.is_mkldnn()) {
        // need to ensure contiguous for non-mkldnn tensors
        input = input.contiguous(backend_memory_format);
        weight = weight.contiguous(backend_memory_format);
        bias = bias.defined() ? bias.contiguous() : bias;
      }
      output = at::mkldnn_convolution(
          input, weight, bias, params.padding, params.stride, params.dilation, params.groups);
```

Reply:
We need this conversion,  because mkldnn_convolution has been exposed to the python side, user can directly call **torch.mkldnn_convolution**, see https://github.com/pytorch/pytorch/issues/73193.